client/elements/text/sc-segmented-text.html

```diff
@@ -550,22 +579,179 @@ <h3>Could not find the sutta text.</h3>
+              let section = firstSection;
+              let toSection = this.shadowRoot.getElementById(idTo);
+              if (toSection) {
+                  toSection = this._getElememtToHighlight(toSection);
+              }
+              const SIDEBYSIDE = !!(this.chosenTextView === 'sidebyside' && this.translatedSutta)
+              while (true) {
+                  if (!section) {
+                      break;
+                  }
+                  let wrapper
+                  let wrapperAlreadyExists = false;
+
+                  if (SIDEBYSIDE) {
+                      [wrapperAlreadyExists, wrapper] = this._putSectionIntoWrapper();
+                  }
+
+                  if (!SIDEBYSIDE) {
+                      section.classList.add('highlight');
+                  }
+
+                  const margin = (section.currentStyle || window.getComputedStyle(section)).margin;
+                  if (section === toSection || !toSection) {
+                      this._processLastSection(section, wrapper, idFrom, idTo, SIDEBYSIDE, wrapperAlreadyExists);
+                      break;
+                  } else if (margin && margin !== '0px') {
+                      section.style.margin = '0';
+                      section.style.padding = margin;
+                  }
+
+                  if (section.nextElementSibling === null && section.parentNode.nodeName === 'P') {
+                      this._processParagraphHighlight(section, SIDEBYSIDE, wrapperAlreadyExists, wrapper);
+                  } else {
+                      let newSection = section.nextElementSibling;
+                      if (SIDEBYSIDE && !wrapperAlreadyExists) {
+                          section.remove();
+                      }
+                      section = newSection;
+                  }
+                  if (SIDEBYSIDE) {
+                      this._addInlineParagraphListeners(wrapper);
+                  }
+              }
+              return firstSection;
+          }
+
           // Scrolls to the chosen section
           _scrollToSection(sectionId, isSmooth, delay) {
-              if (!sectionId) return;
-              try {
-                  const section = this.shadowRoot.getElementById(sectionId);
-                  if (!section) return;
-                  setTimeout(() => {
-                      section.scrollIntoView({
-                          behavior: isSmooth ? 'smooth' : 'instant',
-                          block: 'start',
-                          inline: 'nearest'
-                      });
-                  }, delay);
-              } catch (e) {
-                  console.error(e);
+              if (!sectionId) {
+                  return;
+              }
+              this._removeHighlights();
+              if (sectionId && sectionId.includes('.') &&
+                  (this.chosenTextView === 'none' || !this.translatedSutta)) {
+                  return;
               }
+              setTimeout(() => {
```

Comment:
not `requestAnimationFrame`?

Reply:
no, we need delay because of scrolling to top